x/plans/keeper/migrations.go

```diff
@@ -26,37 +26,44 @@ func (m Migrator) Migrate2to3(ctx sdk.Context) error {
 	return nil
 }
 
+// Migrate3to4 implements protobuf migration from v3 to v4:
+// Now the plan protobuf has a plan policy
 func (m Migrator) Migrate3to4(ctx sdk.Context) error {
 	planIndices := m.keeper.GetAllPlanIndices(ctx)
 
 	for _, planIndex := range planIndices {
-		var oldPlanStruct v3.Plan
-		if found := m.keeper.plansFS.FindEntry(ctx, planIndex, uint64(ctx.BlockHeight()), &oldPlanStruct); !found {
-			return fmt.Errorf("could not find plan with index %s", planIndex)
-		}
+		blocks := m.keeper.plansFS.GetAllEntryVersions(ctx, planIndex, true)
+		for _, block := range blocks {
+			var oldPlanStruct v3.Plan
+			if found := m.keeper.plansFS.FindEntry(ctx, planIndex, block, &oldPlanStruct); !found {
```

Comment:
Use ReadEntry() when we know the specific block version

Reply:
for this migration it is not really needed, also we might want to create such function

Reply:
Already created in CNS-385, but not merged yet.
I will make this change part of CNS-385.